docs/THE_INSIDE_STORY.md

```diff
@@ -0,0 +1,186 @@
+
+Once the setup is complete, Cognito can be configured to use the verified email
+address as the "FROM Email Address" by editing "Email" on the Messaging tab in
+the User Pool. An optional "FROM Sender Name" can also be configured, such as
+"New Jersey Call Centers <callcenters@innovation.nj.gov>".
+
+Special thanks to the New Jersey Cyber Communication & Integration Cell (NJCCIC)
+and AWS support engineers who helped us figure out how to set this up.
+
+## Setting up SMS
+
+Getting SMS configured required a number of steps, lead time, and also some
+bug-dodging trickery.
+
+### Allowing SMS for MFA without shooting yourself in the foot
+
+As it turns out, at least as of this writing in early December 2023, if a User
+Pool wants to allow both MFA methods (Authenticator apps + SMS message)
+utilizing the Cognito Hosted UI, you __*should not*__ configure the "SMS
+message" option while in the User Pool creation flow. If "SMS message" is
+selected, then on the following creation flow page you will see that
+`phone_number` is one of the user "Required Attributes". When the phone number
+is a required attribute, it becomes impossible for the user to choose SMS as
+their MFA method. Only when `phone_number` is not required does Cognito
+correctly prompt the user to choose SMS or Authenticator App during their first
+login. At least this is how it was for our system which does not allow
+self-registration but does not necessarily know phone numbers when a user is
+created.
+
+This seems pretty obviously like a bug in Cognito, but while I did try to urge a
+support engineer to submit a bug report for me when I encountered this issue, I
+am not confident that one was opened.
+
+### An originating phone number
+
+Cognito links to
+[this documentation](https://docs.aws.amazon.com/sns/latest/dg/channels-sms-originating-identities.html)
+for creating an originating identity. We use a Toll-Free Origination Number.
+Fortunately for me, I did not set this piece up, but then unfortunately for this
+documentation, I don't have a summary of any hurdles that had to be overcome
+here.
+
+### Enabling SMS
+
+Cognito links to
+[this documentation](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-sms-settings.html)
+for setting up SMS for Cognito. In practice, it primarily involved following the
+steps which Cognito suggests on the *Messaging* tab of the User Pool: moving SNS
+out of Sandbox mode (just like SES) and requesting a limit increase (or a few
+increases?) to SNS and possibly also Pinpoint and Pinpoint SMS.
+
+Something that was a bit surprising was that even after following all the steps
+and getting SMS correctly setup, there is still an *Info* box on the SMS part of
+the Messaging page with advice for the steps necessary to set up SMS. For this
+confusing UI, I believe an AWS support engineer did submit a request to change
+it so the info box goes away when everything is correctly configured.
+
+Where to configure the base URL
```

Comment:
should this line be formatted as a `##` title (then adjusting the below title too)?

Reply:
Oooh good catch, I think this was a note from me while writing the doc, should be removed now! TY!